src/scene/graph/mod.rs

```diff
@@ -1353,43 +1358,48 @@ impl Graph {
 
     /// Tries to borrow a node using the given handle, fetch its script and cast it to the specified type.
     #[inline]
-    pub fn try_get_script_of<T>(&self, node: Handle<Node>) -> Option<&T>
+    pub fn try_get_script_of<T>(&self, node: Handle<Node>, index: usize) -> Option<&T>
```

Comment:
This method and every other that accepts `T` as a type argument, should search for a first script of type `T`. Similar to Unity's `GetComponent` (https://docs.unity3d.com/ScriptReference/GameObject.GetComponent.html) . Also the doc comment above the method should clearly state that this and other methods search for the first script of the given type.